### DISCUSSION TITLE
Ignore property when mapping using MemberList.Source

### DISCUSSION THREAD

**Discussion by jenscski** [LuckyPennySoftware/AutoMapper#4119] [NA]
### Source/destination types

```csharp
public class Dbo
{
    public string Name { get; set; }

    public string Password { get; set; }
}

public class Dto
{
    public string Name { get; set; }

    public string Password { get; set; }
}
```

### Mapping configuration

```csharp
var config = new MapperConfiguration(cfg => cfg.CreateMap<Dto, Dbo>(MemberList.Source).ForMember(dst => dst.Password, opt => opt.Ignore()));
config.AssertConfigurationIsValid();
```

### Expected behavior

In previous versions this mapping worked as I expected.

### Actual behavior

In version 12.0.0 this mapping fails with the following exception
Is this a change that I'm unable to find in the documentation?

```
Unhandled exception. AutoMapper.AutoMapperConfigurationException:
Unmapped members were found. Review the types and members below.
Add a custom mapping expression, ignore, add a custom resolver, or modify the source/destination type
For no matching constructor, add a no-arg ctor, add optional arguments, or map all of the constructor parameters
===========
Dto -> Dbo (Source member list)
Dto -> Dbo (Source member list)

Unmapped properties:
Password
```

### Steps to reproduce

```csharp
config.AssertConfigurationIsValid();
```

If I add 
```.ForSourceMember(src => src.Password, opt => opt.DoNotValidate())```
 to the code, it will work with version 12.0.0

**Comment by lbargaoanu** [comment]
The new behaviour makes perfect sense to me. The source member is not mapped anymore because you ignored the matching destination member.

**Reply by jenscski** [reply]
The change make sense, I just didn't see it in any release notes / upgrade guide. So was just wondering if it was by intention ...

**Reply by lbargaoanu** [reply]
If it's correct, it's intentional :) I'd rather not spend my time documenting every little change. I guess we'd take a PR for the upgrade guide, but I'm wondering what it would say exactly.

**Reply by lbargaoanu** [reply]
If you want to add a test for this particular case, you can target my branch.
